express-post-json/index.js

```diff
@@ -0,0 +1,26 @@
+const express = require('express');
+const app = express();
+
+app.use(express.json());
+
+let nextId = 1;
+const grades = {};
+
+app.get('/api/grades', (req, res) => {
+  const newArr = [];
+  for (const x in grades) {
+    newArr.push(grades[x]);
+  }
+  res.json(newArr);
+});
+
+app.post('/api/grades', (req, res) => {
+  req.body.id = nextId;
+  grades[nextId] = req.body;
+  res.status(201).json(req.body);
+  nextId++;
```

Comment:
While this will work it is bad practice and can lead to issues. Sending the response should be the last thing to happen within the callback function for the route.

```JAVASCRIPT
nextId++;
res.status(201).json(req.body);
```